File: src/main/java/com/kelsos/mbrc/commands/ReduceVolumeOnRingCommand.java
package com.kelsos.mbrc.commands;

import com.google.inject.Inject;
import com.kelsos.mbrc.data.SocketMessage;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import com.kelsos.mbrc.model.MainDataModel;
import com.kelsos.mbrc.constants.Protocol;
import com.kelsos.mbrc.services.SocketService;

public class ReduceVolumeOnRingCommand implements ICommand {
    private MainDataModel model;
    private SocketService service;

    @Inject public ReduceVolumeOnRingCommand(MainDataModel model, SocketService service) {
        this.model = model;
        this.service = service;
    }

    @Override public void execute(IEvent e) {
        service.sendData(new SocketMessage(Protocol.PlayerVolume, Protocol.Request, (int) (model.getVolume() * 0.2)));
    }
}

File: app/src/main/java/com/kelsos/mbrc/commands/ReduceVolumeOnRingCommand.kt
package com.kelsos.mbrc.commands

import com.kelsos.mbrc.common.state.MainDataModel
import com.kelsos.mbrc.networking.client.SocketMessage
import com.kelsos.mbrc.networking.client.SocketService
import com.kelsos.mbrc.networking.protocol.Protocol
import com.kelsos.mbrc.networking.protocol.ProtocolAction
import com.kelsos.mbrc.networking.protocol.ProtocolMessage

class ReduceVolumeOnRingCommand(
  private val model: MainDataModel,
  private val service: SocketService,
) : ProtocolAction {
  override fun execute(message: ProtocolMessage) {
    if (model.isMute || model.volume == 0) {
      return
    }
    service.sendData(SocketMessage.create(Protocol.PLAYER_VOLUME, (model.volume * 0.2).toInt()))
  }
}
